src/_igraph/graphobject.c

```diff
@@ -2791,6 +2791,67 @@ PyObject *igraphmodule_Graph_LCF(PyTypeObject *type,
   return (PyObject *) self;
 }
 
+/** \ingroup python_interface_graph
+ * \brief Generates a graph with a specified degree sequence
+ * \return a reference to the newly generated Python igraph object
+ * \sa igraph_realize_degree_sequence
+ */
+PyObject *igraphmodule_Graph_Realize_Degree_Sequence(PyTypeObject *type,
+                                 PyObject *args, PyObject *kwds) {
+
+  igraph_vector_t outdeg, indeg;
+  igraph_vector_t *indegp = 0;
+  igraph_edge_type_sw_t allowed_edge_types;
+  igraph_realize_degseq_t method;
+  PyObject *outdeg_o, *indeg_o, *edge_types_o, *method_o;
+  igraphmodule_GraphObject *self;
+  igraph_t g;
+
+  static char *kwlist[] = { "outdeg", "indeg", "allowed_edge_types", "method", NULL };
```

Comment:
The keyword argument ordering is a bit tricky for this function. I agree with your initial assessment that `allowed_edge_types` and `method` should be mandatory; after all, the purpose of this function is to let the user have fine-grained control over how the graph generation happens. However, it bothers me that as a consequence, `indeg` has to be set to `None` explicitly for undirected graphs.

Therefore, I'm leaning towards providing a default value for all the keyword arguments except `outdeg`. I'll do it in a separate commit, just wanted to mention it here in case you disagree.

Reply:
I agree with you, had the same doubt, whatever you choose is going to be just fine for me

Reply:
As defaults, I recommend "simple graph" and "smallest first" method. Simple graphs are the least trivial to construct and the most often needed. "Smallest first" produces connected graphs in the undirected case, which is nice (although everyone else implements "largest first" only).

Reply:
@szhorvat I read your mind yesterday; this is exactly how I set up the defaults.